internal/cmds/cmds.go

```diff
@@ -422,16 +434,89 @@ func copyFiles(ctx log.Logger, fileExtensionSuffix string, extensionSubdirectory
 		}
 	}
 
 	ctx.Log("message", fmt.Sprintf("Migrated %d '%s' files from extension version '%s' to '%s'", numberOfFilesMigrated, fileExtensionSuffix, oldExtensionVersion, newExtensionVersion))
 
+	return fileNamesMigrated, nil
+}
+
+// This need to be only executed by Update operation
+func createDummyStatusFilesIfNeeded(ctx log.Logger, mrseqFilesNameList *list.List) error {
+	if mrseqFilesNameList == nil || mrseqFilesNameList.Len() <= 0 {
+		return nil
+	}
+
+	// Create dummy status file for .mrseq file if status file is not available.
+	newExtensionDirectory := os.Getenv(constants.ExtensionPathEnvName)
+	statusFileDirectoryPath := filepath.Join(newExtensionDirectory, constants.StatusFileDirectory)
+
+	var mrSeqFileName string
+	var mrSeqFileFullPath string
+	var extensionName string
+	var mrSeqFileExtensionIndex int
+	var statusFileName string
+	var statusFilePath string
+	for mreSeqFileNameElement := mrseqFilesNameList.Front(); mreSeqFileNameElement != nil; mreSeqFileNameElement = mreSeqFileNameElement.Next() {
+		mrSeqFileName = (mreSeqFileNameElement.Value).(string)
+
+		// Read the most recently executed sequence number from the .mrseq file
+		mrSeqFileFullPath = filepath.Join(newExtensionDirectory, mrSeqFileName)
+		content, err := os.ReadFile(mrSeqFileFullPath)
+		if err != nil {
+			ctx.Log("error", fmt.Sprintf("Reading mrseq (Most Recently executed Sequence number) from file '%s' failed with error '%s'", mrSeqFileFullPath, err.Error()))
+			return err
+		}
+
+		var mrseqNumber int
+		if content != nil {
+			mrseqNumberString := string(content)
+			mrseqNum, errr := strconv.Atoi(mrseqNumberString)
+			if errr != nil {
+				ctx.Log("error", fmt.Sprintf("mrseqNumberString to mrseqNumber conversion (string to int) of '%s' failed with error '%s'", mrseqNumberString, errr.Error()))
+				return errr
+			} else {
+				mrseqNumber = mrseqNum
+			}
+
+		} else {
+			errorMessage := fmt.Sprintf("Empty .mrseq file content. No sequence number was found inside file  '%s' ", mrSeqFileFullPath)
+			ctx.Log("error", errorMessage)
+			return errors.New(errorMessage)
+		}
+
+		// Find extension name from the .mrseq file
+		mrSeqFileExtensionIndex = strings.Index(mrSeqFileName, constants.MrSeqFileExtension)
+		if mrSeqFileExtensionIndex == -1 {
+			return errors.New(fmt.Sprintf("Invalid mrseq file '%s'", mrSeqFileName))
+		}
+		extensionName = mrSeqFileName[0:mrSeqFileExtensionIndex]
+
+		// Determine status file name and status file path
+		statusFileName = fmt.Sprintf("%s.%d.status", extensionName, mrseqNumber)
+		statusFilePath = filepath.Join(statusFileDirectoryPath, statusFileName)
+
+		// If status file path does not exist, create a dummy status file to prevent poll status timeouts for already executed Run Commands after upgrade.
+		if !handlersettings.DoesFileExist(statusFilePath) {
+			statusReport := types.NewStatusReport(types.StatusSuccess, "Enable", "The script has been executed. However, the real execution state, output, error are unknown.")
```

Comment:
Probably should be more specific and state that the previous status file was deleted and we've replaced it but without the execution state, output, and error.

Reply:
These are the only states CRP understands. So, using "Warning" state now. CRP displays "warning" in instanceView, but it treats as ProivisioningState = Succeeded at the RC ARM resource level.

    // State of an extension as reported by the extension handler.
    public enum ExtensionStates
    {
        Transitioning,
        Error,
        Success,
        Warning,
    }